src/transformers/models/opt/modeling_opt.py

```diff
@@ -647,6 +867,9 @@ def forward(
                 f"The provided attention mask has length {attention_mask.shape[1]}, but its length should be "
                 f"{mask_seq_length} (sum of the lengths of current and past inputs)"
             )
+
+        padding_mask = attention_mask
+
         causal_attention_mask = self._prepare_decoder_attention_mask(
```

Comment:
In that case we need to pass `padding_mask = attention_mask` and we check for the ones array in the `_flash_attention_forward`. If we use any `if-else` condition here it gives an error for the `torch.fx` tests.